File: tests/compile-fail/ForeignPrivateDefaultPropertyCollision.kt
import CompileFailForeignPrivateDefault.IPrivateDefaultProperty

open class ForeignPrivateDefaultPropertyCollision : IPrivateDefaultProperty {
    // An open function is a CLR virtual MethodDef and captures IPropertySlot.get_Value unless the inherited default
    // receives a class-level explicit MethodImpl. Its selected body is private in another assembly.
    open fun get_Value(): Int = 99
}

File: tests/compile-fail/ForeignPrivateDefaultPropertyCollision.cs
namespace CompileFailForeignPrivateDefault;

public interface IPropertySlot
{
    int Value { get; }
}

// C# emits the explicit interface accessor as a private final DIM body. A consumer can inherit it through this
// interface, but another assembly cannot call that body to manufacture a new class-level MethodImpl trampoline.
public interface IPrivateDefaultProperty : IPropertySlot
{
    int IPropertySlot.Value => 42;
}
